sql/analyzer/process_truncate.go

```diff
@@ -0,0 +1,156 @@
+package analyzer
+
+import (
+	"strings"
+
+	"github.com/dolthub/vitess/go/vt/sqlparser"
+
+	"github.com/dolthub/go-mysql-server/sql"
+	"github.com/dolthub/go-mysql-server/sql/plan"
+)
+
+func processTruncate(ctx *sql.Context, a *Analyzer, n sql.Node, scope *Scope) (sql.Node, error) {
+	span, _ := ctx.Span("processTruncate")
+	defer span.Finish()
+
+	deletePlan, ok := n.(*plan.DeleteFrom)
+	if ok {
+		return deleteToTruncate(ctx, a, deletePlan)
+	}
+	truncatePlan, ok := n.(*plan.Truncate)
+	if ok {
+		_, err := validateTruncate(ctx, a, truncatePlan.DatabaseName(), truncatePlan.Child())
+		if err != nil {
+			return nil, err
+		}
+		return truncatePlan, nil
+	}
+	return n, nil
+}
+
+func deleteToTruncate(ctx *sql.Context, a *Analyzer, deletePlan *plan.DeleteFrom) (sql.Node, error) {
+	tbl, ok := deletePlan.Child.(*plan.ResolvedTable)
+	if !ok {
+		return deletePlan, nil
+	}
+	tblName := strings.ToLower(tbl.Name())
+
+	// auto_increment behaves differently for TRUNCATE and DELETE
+	for _, col := range tbl.Schema() {
+		if col.AutoIncrement {
+			return deletePlan, nil
+		}
+	}
+
+	dbName := ""
+	tblFound := false
+	//TODO: if multiple dbs have a table with the same name it's ambiguous, find a way to distinguish (will remove this pass)
+	for _, db := range a.Catalog.AllDatabases() {
+		dbTblNames, err := db.GetTableNames(ctx)
+		if err != nil {
+			return nil, err
+		}
+		for _, dbTblName := range dbTblNames {
+			if strings.ToLower(dbTblName) == tblName {
+				if tblFound == false {
+					tblFound = true
+					dbName = db.Name()
+				} else {
+					return deletePlan, nil
+				}
+			}
+		}
+	}
+	if !tblFound {
+		return deletePlan, nil
+	}
+	dbNameLower := strings.ToLower(dbName)
+
+	for _, db := range a.Catalog.AllDatabases() {
```

Comment:
You should determine which DB the table is in above and use it here, not loop over all DBs.